pkg/providers/cloudstack/decoder/decoder.go

```diff
@@ -16,10 +17,53 @@ const (
 	EksaCloudStackHostPathToMount         = "EKSA_CLOUDSTACK_HOST_PATHS_TO_MOUNT"
 	defaultVerifySslValue                 = "true"
 	CloudStackGlobalAZ                    = "global"
+
+	APIKeyKey    = "api-key"
+	SecretKeyKey = "secret-key"
+	APIUrlKey    = "api-url"
+	VerifySslKey = "verify-ssl"
 )
 
-// ParseCloudStackSecret parses the input b64 string into the ini object to extract out the api key, secret key, and url.
-func ParseCloudStackSecret() (*CloudStackExecConfig, error) {
+// ParseCloudStackCredsFromSecrets parses a list of secrets to extract out the api keys, secret keys, and urls.
+func ParseCloudStackCredsFromSecrets(secrets []apiv1.Secret) (*CloudStackExecConfig, error) {
+	var cloudstackProfiles []CloudStackProfileConfig
```

Comment:
can we set the size here if we know it?

Reply:
that will mean we can no longer use append, right? Instead we have to set using slices? What's the purpose? It seems like that introduces more risk of some off-by-one out of bounds error in the future